### PR TITLE
[EC-135] Delay decryption of provider keys

### PR DESCRIPTION
**Pull request by eliykat**
## Type of change

- [ ] Bug fix
- [ ] New feature development
- [ ] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other

## Code changes

- **file.ext:** Description of what was changed and why

## Before you submit

- [ ] I have checked for **linting** errors (`npm run lint`) (required)
- [ ] I have added **unit tests** where it makes sense to do so (encouraged but not required)
- [ ] This change requires a **documentation update** (notify the documentation team)
- [ ] This change has particular **deployment requirements** (notify the DevOps team)

### CODE REVIEW

File: common/src/interfaces/encryptedOrganizationKeyStore.ts [hunk 1]
```diff
@@ -0,0 +1,4 @@
+export interface EncryptedOrganizationKeyStore {
+  key: string;
+  providerId?: string;
+}
```
Comment (Hinton) [inline common/src/interfaces/encryptedOrganizationKeyStore.ts:4]: What do you think of splitting this into two classes.

```ts
class EncryptedOrganizationKey {
	constructor(private key) {}

  unpack(cryptoService) {
    return cryptoService.rsaDecrypt(this.key);
  }
}
```

```ts
class ProviderEncryptedOrganizationKey {
	constructor(private orgKey, private providerKey) {}

  unpack(cryptoService) {
    return cryptoService.decryptToBytes(new EncString(orgKey), providerKey)
  }
}
```

File: common/src/models/domain/account.ts [hunk 2]
```diff
@@ -20,6 +21,7 @@ import { GeneratedPasswordHistory } from "./generatedPasswordHistory";
 import { Policy } from "./policy";
 import { SymmetricCryptoKey } from "./symmetricCryptoKey";
 
+
```
Comment (Hinton) [inline common/src/models/domain/account.ts:24]: We seem to have gotten an accidental newline added here

File: common/src/services/crypto.service.ts [hunk 3]
```diff
@@ -215,20 +219,35 @@ export class CryptoService implements CryptoServiceAbstraction {
...
+    const encOrgKeyStore = await this.stateService.getEncryptedOrganizationKeys();
+    if (encOrgKeyStore == null) {
       return null;
     }
 
     let setKey = false;
 
-    for (const orgId in encOrgKeys) {
+    for (const orgId in encOrgKeyStore) {
       // eslint-disable-next-line
```
[the remark this answers is not in the document]
Comment (eliykat) [reply common/src/services/crypto.service.ts]: Done. eslint doesn't like us calling prototype methods directly, but we can achieve the same result (i.e. iterating over an object's own properties) by using `Object.keys`.

**Comment by Hinton** [comment]
Hi,

We are currently in the process of archiving this repository and moving the code and future development to the [bitwarden/clients](https://github.com/bitwarden/clients) repository. This PR will be closed but feel free to re-open it again on [bitwarden/clients](https://github.com/bitwarden/clients).

**Pull request closed without merging**